### PR TITLE
CRYP-177: Created the db schema for the tag table.

### PR DESCRIPTION
**Pull request by alexbara2000**
I added the tag entity and all dependencies.

### CODE REVIEW

File: packages/common/src/domain/entities/tag.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Entity, JoinColumn, JoinTable, ManyToMany, ManyToOne, PrimaryColumn } from "typeorm";
+import { Transaction } from "./transaction";
+ import { User } from "./user";
+
+ @Entity()
+ export class Tag {
```
Comment (jason-gerard) [inline packages/common/src/domain/entities/tag.ts:6]: To be more specific we should probably prefix the tag with transaction?

```suggestion
 export class TransactionTag {
```

File: packages/common/src/domain/entities/tag.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+     @PrimaryColumn()
+     tagName: string;
+
+     @PrimaryColumn()
+     userId: Number;
```
Comment (jason-gerard) [inline packages/common/src/domain/entities/tag.ts:11]: You don't need to inlcude this here you can reference the user id from the attribute for the primary key

Comment (jason-gerard) [reply packages/common/src/domain/entities/tag.ts:11]: I'm not sure if you are even able to do this though so maybe leave

File: packages/common/src/domain/entities/TransactionTag.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    @PrimaryColumn()
+    userId: number;
+
+    @ManyToOne(() => User, (user) => user.tags)
+    @JoinColumn({ name: "userId", referencedColumnName: "id" })
+    user: User;
+
+    @ManyToMany(() => Transaction)
+    @JoinTable()
+    Transactions: Transaction[];
```
Comment (jason-gerard) [inline packages/common/src/domain/entities/TransactionTag.ts:19]: Lowercase t "transactions"

File: packages/common/src/domain/entities/user.ts [hunk 4]
```diff
@@ -23,4 +24,7 @@ export class User {
 
     @OneToMany(() => Wallet, (wallet) => wallet.user)
     wallets: Wallet[];
+
+    @OneToMany(() => TransactionTag, (transactionTag) => transactionTag.user)
+    tags: TransactionTag[];
```
Comment (jason-gerard) [inline packages/common/src/domain/entities/user.ts:29]: transactionTags

**Review by jason-gerard — APPROVED** [review]
Just some naming stuff you should be able to fix quick

**Review by Andre-Ibrahim — APPROVED** [review]
Looks good, great job!

**Pull request merged**